hack/openshift/fetch-tektoncd-catalog-tasks.sh

```diff
@@ -99,28 +102,44 @@ EOF
 
 
 get_tasks() {
-  local dest_dir="$1"; shift
-  local catalog="$1"; shift
-  local catalog_version="$1"; shift
-  local -n tasks=$1; shift
-  local type="$1"
+  local dest_dir="$1"; shift || true
+  local catalog="$1"; shift || true
+  local catalog_version="$1"; shift || true
+  local type="$1"; shift || true
+  local -n tasks="$1"
+  local -n stepactions="$1"
+
   info "Downloading tasks from catalog $catalog to $dest_dir directory"
-  for t in ${!tasks[@]} ; do
-    # task filenames do not follow a naming convention,
-    # some are taskname.yaml while others are taskname-task.yaml
-    # so, try both before failing
-    if [[ "$type" == "ecosystem" ]]; then
-      local task_url="$catalog/$catalog_version/tasks/$t/${tasks[$t]}/${t}.yaml"
-    elif [[ "$type" == 'default' ]];then
-      local task_url="$catalog/$catalog_version/task/$t/${tasks[$t]}/${t}.yaml"
-    fi
-
-    mkdir -p "$dest_dir/$t/"
-    local task_path="$dest_dir/$t/$t-task.yaml"
-
-    download_task  "$task_path" "$task_url"  ||
-      die 1 "Failed to download $t"
-  done
+
+  if [[ "$type" == "ecosystem_tasks" || "$type" == 'default' ]]; then
```

Comment:
i dont understand the difference between if and else, i see we are just setting different path and url, that can be handle without the outer if else, i dont feel the need of nested if else

Reply:
The reason for if else is to add tasks and stepactions under respective folders

Reply:
we are taking the same arg input in different variable, then doing nested if else for different variable, i dont get this logic.